homeassistant/components/recorder/history.py

```diff
@@ -92,10 +100,16 @@ def get_significant_states_with_session(
     thermostat so that we get current temperature in our graphs).
     """
     timer_start = time.perf_counter()
-
-    baked_query = hass.data[HISTORY_BAKERY](
-        lambda session: session.query(*QUERY_STATES)
+    need_attributes = (
+        entity_ids is None
+        or not minimal_response
+        or any(
+            split_entity_id(ent_id)[0] in NEED_ATTRIBUTE_DOMAINS
+            for ent_id in entity_ids
+        )
     )
```

Comment:
This can't be deterministic and needs to be set only when requested instead since we need to be backwards compat